utils/zerovec/src/map/vecs.rs

```diff
@@ -547,6 +548,158 @@ where
     }
 }
 
+impl<'a> ZeroVecLike<usize> for FlexZeroVec<'a>
+{
+    type GetType = [u8];
+    type BorrowedVariant = FlexZeroSlice;
+
+    fn zvl_new() -> Self {
+        Self::new()
+    }
+    fn zvl_binary_search(&self, k: &usize) -> Result<usize, usize>
+    {
+        FlexZeroSlice::binary_search(self, *k)
+    }
+    fn zvl_binary_search_in_range(&self, k: &usize, range: Range<usize>) -> Option<Result<usize, usize>>
+    {
+        FlexZeroSlice::binary_search_in_range(&self, *k, range)
+    }
+    fn zvl_binary_search_by(
+        &self,
+        mut predicate: impl FnMut(&usize) -> Ordering,
+    ) -> Result<usize, usize> {
+        FlexZeroSlice::binary_search_by(self, |probe| predicate(&probe))
+    }
+    fn zvl_binary_search_in_range_by(
+        &self,
+        predicate: impl FnMut(&usize) -> Ordering,
+        range: Range<usize>,
+    ) -> Option<Result<usize, usize>> {
+        FlexZeroSlice::binary_search_in_range_by(&self, |probe| predicate(&probe), range)
+    }
+    fn zvl_get(&self, index: usize) -> Option<&[u8]> {
+        self.get_chunk(index)
+    }
+    fn zvl_len(&self) -> usize {
+        FlexZeroSlice::len(self)
+    }
+    fn zvl_is_ascending(&self) -> bool
+    {
+        // FIXME
+        todo!()
+    }
+
+    fn zvl_as_borrowed(&self) -> &FlexZeroSlice {
+        &*self
+    }
+
+    #[inline]
+    fn zvl_get_as_t<R>(g: &[u8], f: impl FnOnce(&usize) -> R) -> R {
+        f(&crate::chunk_to_usize(g, g.len()))
+    }
+}
+
+impl ZeroVecLike<usize> for FlexZeroSlice
+{
+    type GetType = [u8];
+    type BorrowedVariant = FlexZeroSlice;
+
+    fn zvl_new() -> FlexZeroSlice
+    {
+        unreachable!()
+    }
```

Comment:
@Manishearth How do I get this `where Self: Sized` trait function to compile here? What is the right syntax? No matter whether I add `where Self: Sized` here or not, it says "doesn't have a size known at compile-time"

Reply:
I fixed it by removing the method from the trait.

Reply:
For reference: it can be done on stable with an unused HRTB, see https://github.com/rust-lang/rust/issues/48214#issuecomment-1150463333.